tq/meter.go

```diff
@@ -215,14 +218,21 @@ func (m *Meter) str() string {
 	// (%d of %d files, %d skipped) %f B / %f B, %f B skipped
 	// skipped counts only show when > 0
 
+	current := int64(0)
```

Comment:
Since `TransferBytes` ought to happen less frequently than sending an update to the terminal, do you think we could store (in addition to the map of OID -> bytes) a total `currentBytes` in the way that we used to? If we use `atomic` to compute this, I think that it will be a net performance gain by calling `atomic.Load<T>` instead of locking a mutex for the total time it takes to loop over the map.

Reply:
Right, but computing it in a 2nd place is what lead to the bug in the first place. Maybe it means the way that `tools.BodyWithCallback` updates the `tq.Meter` needs to change. I thought about replacing the func callback with a channel, which _should_ ensure the updates are in order.

Reply:
I see; thanks for explaining. It seems like the main change that is required here are some stricter guarantees about the order in which `tools.BodyWithCallback` calls the callback, so perhaps we could guard that with a `sync/mutex`?

I don't think that we have control about the order in which updates are sent to the `tools.BodyWithCallback`, but we can definitely control when the subsequent calls are made.

That all said, I am less concerned about whether we do math once or `n` times, but I think the semantics of trust regarding the `tools.BodyWithCallback` are important for other callsites besides the `tq/meter` one.